Improve structure, readability and reduce repetitiveness in code

FarbodSalamat-Zadeh opened the pull request:
Resolves #263 

This PR improves code style, uses more methods, loops, and greatly improves the code in terms of readability, structure, and reducing repetitiveness.

In addition to reducing the lines of code by around ~270~ 500 lines, more importantly, this will make it much easier to debug, add on to, and maintain certain parts of the code in the future.

Note that this also includes some changes I had made earlier to improve the code style in #249.

FarbodSalamat-Zadeh commented:
One question I had (which I've also annotated with a `TODO` comment in my code) is at [around line 570 of `SelectFeaturesActivity.java`](https://github.com/systers/powerup-android/blob/develop/PowerUp/app/src/main/java/powerup/systers/com/SelectFeaturesActivity.java#L571).

The first part of the old code reads this:

```java
    if (hatPurchased != 0) {
        ...
        SessionHistory.totalPoints = SessionHistory.totalPoints - 
                getmDbHandler().getPointsAccessories(hatPurchased);
        ...
    }
```

But later on in the code, for different accessories, we still use `hatPurchased` for `getPointsAccessories`. For example,


```java
    if (glassesPurchased != 0) {
        ...
        SessionHistory.totalPoints = SessionHistory.totalPoints - 
                getmDbHandler().getPointsAccessories(hatPurchased);
        ...
    }
```

Is this intentional?

I have refactored the code above into a for loop, but I have done so under the assumption that the code above is correct - [see my TODO comment for details](https://github.com/systers/powerup-android/pull/272/files#diff-f0c795ce901d24208f7734b019de946dR432).

FarbodSalamat-Zadeh commented:
A couple changes came to find following @chhavip 's comment in the related issue about the `Utils` class idea - I realised I hadn't used it as much as I had originally planned to, so a couple additional commits optimise the code even more in this way.

FarbodSalamat-Zadeh commented:
I've merged the conflicts in the files just now as well.

chhavip commented:
@FarbodSalamat-Zadeh Good work has been done in this PR but it contains too many commits. These need to be squashed. Please follow https://github.com/chhavip/Git-Guide for help on the same if you are using command line.

chhavip commented:
Can also ask here for more help if that is not sufficient

FarbodSalamat-Zadeh commented:
Would you like me to squash all of the commits from this PR into 1, or to group and squash related commits? If it's the latter, could you give me more guidance, please?

aanchal07 commented:
Please squash them all into 1.

aanchal07 commented on PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java:
```diff
@@ -24,233 +24,186 @@
+            if (mScene.getNextScenarioId() == -1) {
+                Intent myIntent = new Intent(GameActivity.this, GameOverActivity.class);
                 finish();
                 startActivityForResult(intent, 0);
```
change intent to myIntent

FarbodSalamat-Zadeh replied:
Done - I didn't notice that when renaming some of the variables, thanks.

FarbodSalamat-Zadeh commented:
@chhavip @aanchal07 Thank you for your reviews and guidance.

As I mentioned, I've renamed the `intent` variable to `myIntent` to fix errors.

I have also squashed all commits from this PR into one, and then resolved conflicts.

Let me know what you think. Thanks.

chhavip commented:
There are still two commits in this PR, please see if they can be squashed or rebased.

FarbodSalamat-Zadeh commented:
@anubhakushwaha I've resolved the issues you mentioned in the GCI task comment.

FarbodSalamat-Zadeh commented:
@chhavip I've squashed commits again, but there are conflicts that need to be resolved again.

aanchal07 commented:
![image](https://cloud.githubusercontent.com/assets/13906284/21889794/aae9cb5c-d8f0-11e6-87c6-e51920da394b.png)

Please look into this error.

anubhakushwaha commented:
@FarbodSalamat-Zadeh Great,will check it again :)

chhavip commented:
Also is this is GCI task?

aanchal07 commented:
@chhavip Yes its a GCI task "Resolve an open issue from issue list".

anubhakushwaha commented on PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java:
```diff
@@ -19,227 +24,186 @@
+    private void setupListView() {
+        mListAdapter = new ArrayAdapter<>(this, R.layout.simplerow, new ArrayList<String>());
+
+        ListView mainListView = (ListView) findViewById(R.id.mainListView);
```
Move lines 79-81 in setupLayout() so that mListAdapter is initialized before calling updateScenario() else it will give runtime error of mListadapter being a null object on which various functions are being called in updateQA()

FarbodSalamat-Zadeh replied:
Done.

anubhakushwaha commented on PowerUp/app/src/main/res/layout-land/game_activity.xml:
```diff
@@ -52,32 +51,33 @@
-        android:layout_marginTop="10dp"
-        android:id="@+id/relativeLayout">
+        android:layout_marginTop="@dimen/gameActivity_rl_marginTop">
+
```
Change all the id's related to face/hair/cloth/eye to the common id name used for inflation in UiUtils class else it will not be able to inflate and is giving a runtime error.

FarbodSalamat-Zadeh replied:
Done.

anubhakushwaha commented:
@FarbodSalamat-Zadeh I reviewed your code again and suggested few changes,please try incorporating them to avoid app-crash.

FarbodSalamat-Zadeh commented:
Thanks for the feedback, @anubhakushwaha .

I'll do my best to improve the code by the end of today

FarbodSalamat-Zadeh commented:
@anubhakushwaha I've made the requested changes.

I also had to resolve conflicts afterwards - I'm guessing there were some changes made on the develop branch in the last week or so.

Thank you for your help, Anubha - please do let me know what you think, and if there is anything that can be improved.

anubhakushwaha commented on PowerUp/app/src/main/res/layout/game_activity.xml:
```diff
@@ -182,7 +182,7 @@
+            android:id="@+id/clothView"
             android:layout_width="@dimen/cloth_view_width"
             android:layout_height="@dimen/cloth_view_height"
             android:layout_alignTop="@+id/eyeImageView"
```
Please change here as well to just eyeView

anubhakushwaha commented on PowerUp/app/src/main/res/layout/game_activity.xml:
```diff
@@ -182,7 +182,7 @@
             android:src="@drawable/hair1" />
```
Here also change to just faceView

anubhakushwaha commented on PowerUp/app/src/main/res/layout-land/game_activity.xml:
```diff
@@ -79,14 +79,14 @@
+            android:id="@+id/eyeView"
             android:layout_width="@dimen/eye_view_width"
             android:layout_height="@dimen/eye_view_height"
             android:layout_alignBottom="@id/faceImageView"
```
please change to faceView

anubhakushwaha commented on PowerUp/app/src/main/res/layout-land/game_activity.xml:
```diff
@@ -95,14 +95,14 @@
+            android:id="@+id/clothView"
             android:layout_width="@dimen/cloth_view_width"
             android:layout_height="@dimen/cloth_view_height"
             android:layout_alignTop="@+id/eyeImageView"
```
here as well

FarbodSalamat-Zadeh commented:
@anubhakushwaha I've made the changes - sorry, I must have missed those off in the previous commit.

Let me know what you think.

anubhakushwaha commented:
@FarbodSalamat-Zadeh Could you please squash the commits

aanchal07 commented:
Conflicts in the PR.
